### ISSUE TITLE
Linking libraries issue with v2.5+

### ISSUE THREAD

**Issue by guiguem** [project8/scarab#57]
Using Scarab version later than 2.5 (since the changes in the PackageBuilder), it seems I cannot link libraries together.
More specifically, I have a set of libraries that need to be linked to a library (CORE_DEP_LIBRARIES), but it seems that these libraries cannot be found (aka Misc and ModProb) (see below) leading to 
```
[ 62%] Linking CXX shared library libPTCore.dylib
Undefined symbols for architecture x86_64:
  "GetErecCCQE(double, double)", referenced from:
      TSample::GenerateData(double const&, int&, std::__1::vector<double, std::__1::allocator<double> >&, std::__1::vector<double, std::__1::allocator<double> >&, std::__1::vector<double, std::__1::allocator<double> >&) in TSample.cxx.o
....... 
```
If I use the version 2.4.11, everything works very fine.

Here is my CmakeLists.txt
```
######################
#   Core             #
######################

set (CORE_HEADERFILES
    TApplication.h
    TBin.h
    TBinning.h
    TCholesThrow.h
    TLikelihood.h
    TOscParamHolder.h
    TOscProb.h
    TParListLoader.h
    TPScale.h
    TSample.h
    TSampleFactory.h
    TSplinePoints.h
    TSplineRF.h
)

set (CORE_DEP_LIBRARIES
    ModProb
    Scarab_PTheta
    Misc
)

set (CORE_SOURCEFILES
    TApplication.cxx
    TBin.cxx
    TBinning.cxx
    TCholesThrow.cxx
    TLikelihood.cxx
    TOscParamHolder.cxx
    TOscProb.cxx
    TParListLoader.cxx
    TPScale.cxx
    TSample.cxx
    TSampleFactory.cxx
    TSplinePoints.cxx
    TSplineRF.cxx
)


set( CORE_LINKDEF_HEADERFILE LinkDef/CoreLinkDef.h )
set( CORE_SOURCEFILES ${CORE_SOURCEFILES} ${CMAKE_CURRENT_BINARY_DIR}/PTCoreDict.cxx )
set( CORE_DICT_PCMFILE ${CMAKE_CURRENT_BINARY_DIR}/PTCoreDict_rdict.pcm )

##################################################

ROOT_GENERATE_DICTIONARY(  PTCoreDict ${CORE_DICT_HEADERFILES} LINKDEF ${CORE_LINKDEF_HEADERFILE} )
pbuilder_install_files( ${LIB_INSTALL_DIR} ${CORE_DICT_PCMFILE} )

# ###########
# # Library #
# ###########

pbuilder_library( PTCore CORE_SOURCEFILES CORE_DEP_LIBRARIES )
pbuilder_install_headers (${CORE_HEADERFILES})
```

Temp solution:
I will temporarily revert to 2.4.11.

**Comment by nsoblath** [comment]
I believe I've confirmed this with dripline-cpp.  I had been having an issue during the linking stage when building the examples library, which uses the feature where it links in other libraries from the same build.  I hadn't associated this with the change in the PackageBuilder when I saw it before, but I can confirm that if I put back the old code (from v2.4.11 as @guiguem mentions), it works fine.
